app/controllers/exam_templates_controller.rb

```diff
@@ -103,6 +105,54 @@ def download_generate
               type: "application/pdf")
   end
 
+  def save_cover(exam_template)
+    pdf = CombinePDF.load File.join(exam_template.base_path, exam_template.filename)
+    cover = pdf.pages[0]
+    cover_page = CombinePDF.new
+    cover_page << cover
+    imglist = Magick::Image.from_blob(cover_page.to_pdf) do
+      self.quality = 100
+      self.density = '300'
+    end
+    imglist.first.write(File.join(exam_template.base_path, 'cover.jpg'))
+  end
+
+  def show_cover
+    assignment = Assignment.find(params[:assignment_id])
+    exam_template = assignment.exam_templates.find(params[:id])
+    send_file File.join(exam_template.base_path, 'cover.jpg'), disposition: 'inline', filename: 'cover.jpg'
```

Comment:
Do a bit of error-checking here; if the `cover.jpg` file doesn't exist, send a `head :not_found` or something.